stacker/blueprints/rds/base.py

```diff
@@ -0,0 +1,336 @@
+                Engine=self.ENGINE or Ref("Engine"),
+                EngineVersion=Ref("EngineVersion"),
+                MasterUsername=Ref("MasterUser"),
+                MasterUserPassword=Ref("MasterUserPassword"),
+                MultiAZ=Ref("MultiAZ"),
+                PreferredBackupWindow=Ref("PreferredBackupWindow"),
+                StorageEncrypted=Ref("StorageEncrypted"),
+                VPCSecurityGroups=[Ref(SECURITY_GROUP), ]))
+
+        t.add_resource(
+            DBInstance(
+                DBINSTANCE_WITH_IOPS,
+                Condition="ProvisionedIOPS",
+                AllocatedStorage=Ref("AllocatedStorage"),
+                AllowMajorVersionUpgrade=Ref("AllowMajorVersionUpgrade"),
+                AutoMinorVersionUpgrade=Ref("AutoMinorVersionUpgrade"),
+                BackupRetentionPeriod=Ref("BackupRetentionPeriod"),
+                DBName=Ref("DatabaseName"),
+                DBInstanceClass=Ref("InstanceType"),
+                DBSubnetGroupName=Ref(SUBNET_GROUP),
+                Engine=self.ENGINE or Ref("Engine"),
+                EngineVersion=Ref("EngineVersion"),
+                MasterUsername=Ref("MasterUser"),
+                MasterUserPassword=Ref("MasterUserPassword"),
+                MultiAZ=Ref("MultiAZ"),
+                PreferredBackupWindow=Ref("PreferredBackupWindow"),
+                VPCSecurityGroups=[Ref(SECURITY_GROUP), ],
+                StorageEncrypted=Ref("StorageEncrypted"),
+                StorageType="io1",
+                Iops=Ref("IOPS")))
+
+    def create_dns_records(self):
+        t = self.template
+        endpoint = self.get_db_endpoint()
+
+        # Setup CNAME to db
+        t.add_resource(
+            RecordSetType(
+                DNS_RECORD,
+                # Appends a "." to the end of the domain
+                HostedZoneId=Ref("InternalZoneId"),
+                Comment="RDS DB CNAME Record",
+                Name=Join(".", [Ref("InternalHostname"),
+                          Ref("InternalZoneName")]),
+                Type="CNAME",
+                TTL="120",
+                ResourceRecords=[endpoint],
+                Condition="CreateInternalHostname"))
+
+    def create_db_outputs(self):
+        t = self.template
+        t.add_output(Output("DBAddress", Value=self.get_db_endpoint()))
+        t.add_output(Output("DBInstance", Value=Ref(self.get_db_instance())))
+        t.add_output(
+            Output(
+                "DBCname",
+                Condition="CreateInternalHostname",
+                Value=Ref(DNS_RECORD)))
+
+    def create_template(self):
+        self.create_conditions()
+        self.create_subnet_group()
+        self.create_security_group()
+        self.create_rds()
+        self.create_dns_records()
+        self.create_db_outputs()
+
+
+class ReadReplica(MasterInstance):
+    """ Blueprint for a Read replica RDS Database Instance. """
+    def _get_parameters(self):
+        parameters = common_parameters()
```

Comment:
There are parameters that come from the MasterInstance that shouldn't be used in ReadReplicas. A better design might be to have a BaseInstance type or something, and have MasterInstance & ReadReplica both subclass off of it, rather than ReadReplica being based on MasterInstance.